python/private/py_executable.bzl

```diff
@@ -126,6 +126,24 @@ Optional; the name of the source file that is the main entry point of the
 application. This file must also be listed in `srcs`. If left unspecified,
 `name`, with `.py` appended, is used instead. If `name` does not match any
 filename in `srcs`, `main` must be specified.
+
+This is mutually exclusive with {obj}`main_module`.
+""",
+        ),
+        "main_module": lambda: attrb.String(
```

Comment:
nit: what about `main` becoming `main_module` if it cannot find the file in `srcs`?

That said, then we would have worse error messages, so this solution is also good.

Reply:
I'd rather not because it's not clear how to transform the file name into a module name.